azure/scope/machine_test.go

```diff
@@ -407,19 +409,29 @@ func TestMachineScope_RoleAssignmentSpecs(t *testing.T) {
 						RoleAssignmentName: "azure-role-assignment-name",
 					},
 				},
+				ClusterScoper: &ClusterScope{
+					AzureCluster: &infrav1.AzureCluster{
+						Spec: infrav1.AzureClusterSpec{},
+					},
+				},
 			},
-			want: []azure.RoleAssignmentSpec{
-				{
+			want: []azure.ResourceSpecGetter{
+				&roleassignments.RoleAssignmentSpec{
+					ResourceType: azure.VirtualMachine,
 					MachineName:  "machine-name",
 					Name:         "azure-role-assignment-name",
-					ResourceType: azure.VirtualMachine,
+					//ToDo(@sonasingh46): Use mock at this place
+					ResourceGroup:    "",
+					Scope:            azure.GenerateScope(""),
+					RoleDefinitionID: azure.GenerateContributorRoleDefinitionID(""),
+					PrincipalID:      to.StringPtr("fakePrincipalID"),
```

Comment:
As we now assign `ResourceGroup `, `Scope `, `RoleDefinitionID ` and `PrincipalID` in the machine scope itself. (azure/scope/machine.go:func (m *MachineScope) RoleAssignmentSpecs(principalID *string) []azure.ResourceSpecGetter ) -- we may need to mock these here but to being with I am passing an empty string.

Reply:
I think we're going to want to set those values in the machine scope so we can make sure the fields are getting populated correctly. Check out [this file](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1939/files#diff-2a71aa3df4645e00fede1673e7b5294c50dd82516ae3f79bbb20ed601e6d9839R1338) for an example on how to do that. Also you're going to have to pass the subscription ID through `ClusterScope.AzureClients.EnvironmentSettings.Values` instead of as a normal machine scope field.